curriculum/challenges/english/20-upcoming-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/64dc97ab25730b3e9de86ea5.md

```diff
@@ -0,0 +1,51 @@
+---
+id: 64dc97ab25730b3e9de86ea5
+title: Step 16
+challengeType: 20
+dashedName: step-16
+---
+
+# --description--
+
+At the end of this project you will create a recursive solution to the Tower of Hanoi puzzle, but now you are going to explore an iterative approach to this problem.
+
+Start by adding a `for` loop to your function that iterates through the number of moves and prints the current iteration number.
+
+# --hints--
+
+You should write a `for` loop to iterate through the number of moves. Use the `range()` function for that.
+
+```js
+({ test: () => assert.match(code, /for\s+.\s+in\s+range\s*\(\s*number_of_moves\s*\)\s*:/ )})
+
+```
+
+You should print the current move number at each iteration.
+
+```js
+({ test: () => assert.match(code, /for\s+(.)\s+in\s+range\s*\(\s*number_of_moves\s*\)\s*:\s+print\s*\(\s*\1\s*\)/ )})
+
+```
```

Comment:
````suggestion
```js
({ test: () => {
    const rgs = [
        /for\s+.\s+in\s+range\s*\(\s*number_of_moves\s*\)\s*:/,
        /for\s+\w+\s+in\s+range\s*\(\s*0\s*,\s*number_of_moves\s*\)\s*:/,
        /for\s+\w+\s+in\s+range\s*\(\s*0\s*,\s*number_of_moves\s*,\s*1\s*\)\s*:/
    ]
    const loop = rgs.some(r => code.match(r));
    assert.isTrue(loop);
  }
})
```

You should print the current move number at each iteration.

```js
({ test: () => {
    const rgs = [
        /for\s+(\w+)\s+in\s+range\s*\(\s*number_of_moves\s*\)\s*:\s+print\s*\(\s*\1\s*\)/,
        /for\s+(\w+)\s+in\s+range\s*\(\s*0\s*,\s*number_of_moves\s*\)\s*:\s+print\s*\(\s*\1\s*\)/,
        /for\s+(\w+)\s+in\s+range\s*\(\s*0\s*,\s*number_of_moves\s*,\s*1\s*\)\s*:\s+print\s*\(\s*\1\s*\)/
    ]
    const loop = rgs.some(r => code.match(r));
    assert.isTrue(loop);
  }
})
```
````